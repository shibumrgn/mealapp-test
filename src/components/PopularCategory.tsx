import React, {useEffect, useState} from 'react';
import Section from './Section';
import axios from 'axios';
import {ICategory} from '../interfaces/ICategory';
import CustomButton from './CustomButton';
import {IRecipe} from '../interfaces/IRecipe';
import SquareRecipe from './recipes/SquareRecipe';

const PopularCategory = () => {
    const [categories, setCategories] = useState<ICategory[]>([]);
    const [selectedCategory, setSelectedCategory] = useState<ICategory>();
    const [categoryRecipes, setCategoryRecipes] = useState<IRecipe[]>([]);

    const _getPopularCategories = async () => {
        const response = await axios.get(
            'https://www.themealdb.com/api/json/v1/1/list.php?c=list',
        );
        if (response?.data?.meals) {
            setCategories(response.data?.meals);
            if (categories?.length > 0) {
                setSelectedCategory(categories[0]);
            }
        }
    };

    const _getCategoryRecipes = async () => {
        const response = await axios.get(
            `https://www.themealdb.com/api/json/v1/1/filter.php?c=${selectedCategory?.strCategory}`,
        );
        if (response?.data?.meals) {
            setCategoryRecipes(response.data?.meals);
        }
    };

    useEffect(() => {
        _getPopularCategories();
    }, []);

    useEffect(() => {
        _getCategoryRecipes();
    }, [selectedCategory]);

    return (
        <>
            <Section title="Popular Category" marginLeft={20}>
                {categories?.map(category => (
                    <CustomButton
                        key={category?.strCategory}
                        title={category?.strCategory}
                        isHighlighted={
                            category?.strCategory ===
                            selectedCategory?.strCategory
                        }
                        onClick={() => setSelectedCategory(category)}
                    />
                ))}
            </Section>
            <Section>
                {categoryRecipes?.map(recipe => (
                    <SquareRecipe key={recipe?.idMeal} recipe={recipe} />
                ))}
            </Section>
        </>
    );
};

export default PopularCategory;
