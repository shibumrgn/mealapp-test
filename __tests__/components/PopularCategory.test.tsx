import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PopularCategory from '../../src/components/PopularCategory';
jest.useFakeTimers();
jest.mock('axios');

it('renders correctly', () => {
    const render = renderer.create(<PopularCategory />)?.toJSON();
    expect(render).toMatchSnapshot();
});
