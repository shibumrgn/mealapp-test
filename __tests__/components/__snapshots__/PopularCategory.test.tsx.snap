// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "marginBottom": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": 20,
          "marginLeft": 20,
        }
      }
    >
      Popular Category
    </Text>
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingLeft": 20,
          "width": "100%",
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>,
  <View
    style={
      Object {
        "marginBottom": 30,
      }
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingLeft": 20,
          "width": "100%",
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>,
]
`;
